### DISCUSSION TITLE
Awesome List of WinForms libraries with available source code

### DISCUSSION THREAD

**Discussion by tbolon** [dotnet/winforms#7501] [Show and tell]
As I was digging into some (old) code and libraries I use for my projects, I decided it was time to put a public list of Winforms libraries / projects. As I could not find one on github I decided to create my own.

<https://github.com/tbolon/awesome-dotnet-winforms>

This list is certainly subjective and can be discussed, but anyway I think it could help newcomers or even winforms veterans. Feel free to contribute 😊.

My only requirement is that source code should be available somewhere, so you should check carefully the licence requirements of each project as it could be not be released under MIT Open Source license.

Also, in case of controls with custom designer, they have not always been updated to support the new Out-Of-Process designer, and should be checked.

**Comment by KlausLoeffelmann** [comment]
This is awesome!!

Thank you SO much for assembling this!
@merriemcgaw FYI!

**Reply by tbolon** [reply]
Thanks Klaus 🙏

**Comment by KlausLoeffelmann** [comment]
It would be ideal to add a comment in the main post description, that not everything might be MIT Open Source, and folks should very carefully check the respective license requirements. And also, that if it is about custom controls whatever Control Designer might be provided in the context those may NOT be working out of the box in the .NET (Core) Out-Of-Process Designer. You could for those cases link to the respective .NET Blog Designer Blog Posts.

**Reply by tbolon** [reply]
I have updated my message. I will check if I can add a badge to the repo with the licence. Feel free to give your advice.

**Reply by kirsan31** [reply]
@tbolon you can update \ replace [System.Windows.Forms.DataVisualization](https://github.com/dotnet/winforms-datavisualization) with my [fork](https://github.com/kirsan31/winforms-datavisualization) (new designer supported and other improvements + fixes).

**Reply by tbolon** [reply]
@kirsan31 I have updated the readme, thanks 👍

**Comment by prahladyeri** [comment]
WinForms is as old as hills but still as relevant and thriving even in 2025. Thank you for creating this!

**Comment by koszeggy** [comment]
May I add [mine](https://github.com/koszeggy/KGySoft.WinForms), too? Feel free to ignore it if it's not as awesome as the others. It's about two decades old, opened to public a few months ago, first public non-prerelease version is coming soon.

Seeing your categories maybe my [drawing libraries](https://github.com/koszeggy/KGySoft.Drawing) worth mentioning as well. Not exclusively for WinForms, but it can be [used](https://github.com/koszeggy/KGySoft.Drawing/tree/master/Examples/WinForms) in a WinForms app as well ([among others](https://github.com/koszeggy/KGySoft.Drawing/tree/master/Examples)).

Similarly, I have a platform-independent [business objects](https://github.com/koszeggy/KGySoft.CoreLibraries/blob/master/README.md#business-objects) and [command/property binding](https://github.com/koszeggy/KGySoft.CoreLibraries/blob/master/README.md#command-binding) engine in a more general library that I regularly use in WinForms projects (see a [demo](https://github.com/koszeggy/KGySoft.ComponentModelDemo) or a [real-life app](https://github.com/koszeggy/KGySoft.Drawing.Tools)).

**Reply by tbolon** [reply]
Feel free to send a PR, I will review and certainly add them!
